### PR TITLE
feat(core): support syncing workspaces and blobs in the background

### PR DESCRIPTION
**Pull request by himself65**

### CODE REVIEW

File: packages/workspace/src/affine/sync.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  rootGuid: string,
+  doc: Doc,
+  signal: AbortSignal
+): Promise<void> => {
+  if (signal.aborted) {
+    return;
+  }
+  const binary = await downloadBinaryFromCloud(rootGuid, doc.guid);
+  if (typeof binary !== 'boolean') {
+    applyUpdate(doc, new Uint8Array(binary), 'affine-cloud-service');
```
Comment (pengx17) [inline packages/workspace/src/affine/sync.ts:21]: I am a little worried that applying update here will cause performance issues for the foreground tasks. is it possible to move to worker instead?

File: packages/workspace/src/atom.ts [hunk 2]
```diff
@@ -175,6 +176,7 @@ const fetchMetadata: FetchMetadata = async (get, { signal }) => {
             version: WorkspaceVersion.DatabaseV3,
           }))
         );
+        Adapter.Events['service:start']?.();
```
Comment (pengx17) [inline packages/workspace/src/atom.ts:179]: shall we do polling every a few minutes?

File: packages/workspace/src/affine/sync.ts (hunk 1 above)
Comment (himself65) [reply packages/workspace/src/affine/sync.ts]: Fixed

File: packages/workspace/src/atom.ts (hunk 2 above)
Comment (himself65) [reply packages/workspace/src/atom.ts]: No, socket.io will handle everything when the connection established

**Pull request merged**